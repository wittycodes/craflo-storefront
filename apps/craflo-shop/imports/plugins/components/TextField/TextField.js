"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

require("core-js/modules/es7.object.get-own-property-descriptors");

require("core-js/modules/web.dom.iterable");

require("core-js/modules/es6.array.iterator");

require("core-js/modules/es6.object.keys");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _core = require("@material-ui/core");

var _utils = require("@material-ui/utils");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

// Styles for the base input component
var useInputStyles = (0, _core.makeStyles)(function (theme) {
  return {
    root: {
      "position": "relative",
      "borderRadius": theme.shape.borderRadius,
      "&:hover $notchedOutline": {
        borderColor: theme.palette.colors.black40
      },
      // Reset on touch devices, it doesn't add specificity
      "@media (hover: none)": {
        "&:hover $notchedOutline": {
          borderColor: theme.palette.colors.black20
        }
      },
      "&$focused $notchedOutline": {
        borderColor: theme.palette.colors.reactionBlue400,
        borderWidth: 2
      },
      "&$error $notchedOutline": {
        borderColor: theme.palette.error.main
      },
      "&$disabled $notchedOutline": {
        borderColor: theme.palette.action.disabled
      },
      "&$disabled $input": {
        backgroundColor: theme.palette.colors.black10
      }
    },

    /* Styles applied to the root element if the component is focused. */
    focused: {},

    /* Styles applied to the root element if `error={true}`. */
    error: {},

    /* Styles applied to the root element if `disabled={true}`. */
    disabled: {},

    /* Styles applied to the `input` element if `margin="dense"`. */
    marginDense: {},

    /* Styles applied to the root element if `startAdornment` is provided. */
    multiline: {
      "padding": 0,
      "backgroundColor": theme.palette.colors.black02,
      "&$marginDense": {
        padding: 0
      }
    },

    /* Styles applied to the `NotchedOutline` element. */
    notchedOutline: {
      borderColor: theme.palette.colors.black20,
      padding: "0"
    },
    input: _objectSpread({
      backgroundColor: theme.palette.colors.black02,
      padding: "11.5px 8px"
    }, theme.typography.body2)
  };
}); // Styles for the label above the field

var useInputLabelStyles = (0, _core.makeStyles)(function (theme) {
  return {
    root: _objectSpread({
      marginBottom: theme.spacing(1)
    }, theme.typography.h5),
    formControl: {
      position: "static",
      left: 0,
      top: 0,
      transform: "none"
    },
    shrink: {
      transform: "none",
      transformOrigin: "top left"
    },
    outlined: {
      "transform": "none",
      "&$shrink": {
        transform: "none"
      }
    }
  };
}); // Styles for the helper text below the field

var useFormHelperTextStyles = (0, _core.makeStyles)(function (theme) {
  return {
    root: _objectSpread(_objectSpread({}, theme.typography.body2), {}, {
      color: theme.palette.colors.black55
    }),
    contained: {
      marginLeft: 0,
      marginRight: 0
    }
  };
});
/**
 * @name TextField
 * @param {Object} props Component props
 * @returns {React.Component} returns a React component
 */

var TextField = _react.default.forwardRef(function TextField(props, ref) {
  var inputClasses = useInputStyles();
  var inputLabelClasses = useInputLabelStyles();
  var formHelperTextClasses = useFormHelperTextStyles();
  return /*#__PURE__*/_react.default.createElement(_core.TextField, (0, _extends2.default)({
    FormHelperTextProps: {
      classes: formHelperTextClasses
    },
    InputProps: {
      classes: inputClasses,
      notched: false
    },
    InputLabelProps: {
      shrink: true,
      classes: inputLabelClasses
    },
    fullWidth: true,
    variant: "outlined"
  }, props, {
    ref: ref
  }));
});
/* eslint-disable react/boolean-prop-naming */

/**
 *
 * The following prop-type definitions are copied from the Material UI TextField component to aide in documentation generation.
 * Source: [@material-ui/core/TextField](https://github.com/mui-org/material-ui/blob/master/packages/material-ui/src/TextField/TextField.js)
 */


TextField.propTypes = {
  /**
   * This prop helps users to fill forms faster, especially on mobile devices.
   * The name can be confusing, as it's more like an autofill.
   * You can learn more about it [following the specification](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill).
   */
  FormHelperTextProps: _propTypes.default.object,

  /**
   * If `true`, the `input` element will be focused during the first mount.
   */
  InputLabelProps: _propTypes.default.object,

  /**
   * @ignore
   */
  InputProps: _propTypes.default.object,

  /**
   * Override or extend the styles applied to the component.
   * See [CSS API](#css) below for more details.
   */
  SelectProps: _propTypes.default.object,

  /**
   * @ignore
   */
  autoComplete: _propTypes.default.string,

  /**
   * The color of the component. It supports those theme colors that make sense for this component.
   */
  autoFocus: _propTypes.default.bool,

  /**
   * The default value of the `input` element.
   */
  children: _propTypes.default.node,

  /**
   * If `true`, the `input` element will be disabled.
   */
  className: _propTypes.default.string,

  /**
   * If `true`, the label will be displayed in an error state.
   */
  classes: _propTypes.default.object,

  /**
   * Props applied to the [`FormHelperText`](/api/form-helper-text/) element.
   */
  color: _propTypes.default.oneOf(["primary", "secondary"]),

  /**
   * If `true`, the input will take up the full width of its container.
   */
  defaultValue: _propTypes.default.any,

  /**
   * The helper text content.
   */
  disabled: _propTypes.default.bool,

  /**
   * @ignore
   */
  error: _propTypes.default.bool,

  /**
   * The id of the `input` element.
   * Use this prop to make `label` and `helperText` accessible for screen readers.
   */
  fullWidth: _propTypes.default.bool,

  /**
   * Props applied to the [`InputLabel`](/api/input-label/) element.
   */
  helperText: _propTypes.default.node,

  /**
   * Props applied to the Input element.
   * It will be a [`FilledInput`](/api/filled-input/),
   * [`OutlinedInput`](/api/outlined-input/) or [`Input`](/api/input/)
   * component depending on the `variant` prop value.
   */
  hiddenLabel: _propTypes.default.bool,

  /**
   * [Attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Attributes) applied to the `input` element.
   */
  id: _propTypes.default.string,

  /**
   * Pass a ref to the `input` element.
   */
  inputProps: _propTypes.default.object,

  /**
   * The label content.
   */
  inputRef: _utils.refType,

  /**
   * If `dense` or `normal`, will adjust vertical spacing of this and contained components.
   */
  label: _propTypes.default.node,

  /**
   * If `true`, a textarea element will be rendered instead of an input.
   */
  margin: _propTypes.default.oneOf(["none", "dense", "normal"]),

  /**
   * Name attribute of the `input` element.
   */
  multiline: _propTypes.default.bool,

  /**
   * @ignore
   */
  name: _propTypes.default.string,

  /**
   * Callback fired when the value is changed.
   *
   * @param {object} event The event source of the callback.
   * You can pull out the new value by accessing `event.target.value` (string).
   */
  onBlur: _propTypes.default.func,

  /**
   * @ignore
   */
  onChange: _propTypes.default.func,

  /**
   * The short hint displayed in the input before the user enters a value.
   */
  onFocus: _propTypes.default.func,

  /**
   * If `true`, the label is displayed as required and the `input` element` will be required.
   */
  placeholder: _propTypes.default.string,

  /**
   * Number of rows to display when multiline option is set to true.
   */
  required: _propTypes.default.bool,

  /**
   * Maximum number of rows to display when multiline option is set to true.
   */
  rows: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.number]),

  /**
   * Render a [`Select`](/api/select/) element while passing the Input element to `Select` as `input` parameter.
   * If this option is set you must pass the options of the select as children.
   */
  rowsMax: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.number]),

  /**
   * Props applied to the [`Select`](/api/select/) element.
   */
  select: _propTypes.default.bool,

  /**
   * The size of the text field.
   */
  size: _propTypes.default.oneOf(["small", "medium"]),

  /**
   * Type of the `input` element. It should be [a valid HTML5 input type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Form_%3Cinput%3E_types).
   */
  type: _propTypes.default.string,

  /**
   * The value of the `input` element, required for a controlled component.
   */
  value: _propTypes.default.any,

  /**
   * The variant to use.
   */
  variant: _propTypes.default.oneOf(["standard", "outlined", "filled"])
};
/* eslint-enable react/boolean-prop-naming */

var _default = TextField;
exports.default = _default;