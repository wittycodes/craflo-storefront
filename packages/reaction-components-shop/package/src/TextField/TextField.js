import React from "react";
import PropTypes from "prop-types";
import {
  TextField as MuiTextField,
  makeStyles
} from "@material-ui/core";
import { refType } from "@material-ui/utils";

// Styles for the base input component
const useInputStyles = makeStyles((theme) => ({
  root: {
    "position": "relative",
    "borderRadius": theme.shape.borderRadius,
    "&:hover $notchedOutline": {
      borderColor: theme.palette.colors.black40
    },
    // Reset on touch devices, it doesn't add specificity
    "@media (hover: none)": {
      "&:hover $notchedOutline": {
        borderColor: theme.palette.colors.black20
      }
    },
    "&$focused $notchedOutline": {
      borderColor: theme.palette.colors.reactionBlue400,
      borderWidth: 2
    },
    "&$error $notchedOutline": {
      borderColor: theme.palette.error.main
    },
    "&$disabled $notchedOutline": {
      borderColor: theme.palette.action.disabled
    },
    "&$disabled $input": {
      backgroundColor: theme.palette.colors.black10
    }
  },
  /* Styles applied to the root element if the component is focused. */
  focused: {},
  /* Styles applied to the root element if `error={true}`. */
  error: {},
  /* Styles applied to the root element if `disabled={true}`. */
  disabled: {},
  /* Styles applied to the `input` element if `margin="dense"`. */
  marginDense: {},
  /* Styles applied to the root element if `startAdornment` is provided. */
  multiline: {
    "padding": 0,
    "backgroundColor": theme.palette.colors.black02,
    "&$marginDense": {
      padding: 0
    }
  },
  /* Styles applied to the `NotchedOutline` element. */
  notchedOutline: {
    borderColor: theme.palette.colors.black20,
    padding: "0"

  },
  input: {
    backgroundColor: theme.palette.colors.black02,
    padding: "11.5px 8px",
    ...theme.typography.body2
  }
}));

// Styles for the label above the field
const useInputLabelStyles = makeStyles((theme) => ({
  root: {
    marginBottom: theme.spacing(1),
    ...theme.typography.h5
  },
  formControl: {
    position: "static",
    left: 0,
    top: 0,
    transform: "none"
  },
  shrink: {
    transform: "none",
    transformOrigin: "top left"
  },
  outlined: {
    "transform": "none",
    "&$shrink": {
      transform: "none"
    }
  }
}));

// Styles for the helper text below the field
const useFormHelperTextStyles = makeStyles((theme) => ({
  root: {
    ...theme.typography.body2,
    color: theme.palette.colors.black55
  },
  contained: {
    marginLeft: 0,
    marginRight: 0
  }
}));

/**
 * @name TextField
 * @param {Object} props Component props
 * @returns {React.Component} returns a React component
 */
const TextField = React.forwardRef(function TextField(props, ref) {
  const inputClasses = useInputStyles();
  const inputLabelClasses = useInputLabelStyles();
  const formHelperTextClasses = useFormHelperTextStyles();

  return (
    <MuiTextField
      FormHelperTextProps={{
        classes: formHelperTextClasses
      }}
      InputProps={{
        classes: inputClasses,
        notched: false
      }}
      InputLabelProps={{
        shrink: true,
        classes: inputLabelClasses
      }}
      fullWidth
      variant="outlined"
      {...props}
      ref={ref}
    />
  );
});

/* eslint-disable react/boolean-prop-naming */
/**
 *
 * The following prop-type definitions are copied from the Material UI TextField component to aide in documentation generation.
 * Source: [@material-ui/core/TextField](https://github.com/mui-org/material-ui/blob/master/packages/material-ui/src/TextField/TextField.js)
 */
TextField.propTypes = {
  /**
   * This prop helps users to fill forms faster, especially on mobile devices.
   * The name can be confusing, as it's more like an autofill.
   * You can learn more about it [following the specification](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill).
   */
  FormHelperTextProps: PropTypes.object,
  /**
   * If `true`, the `input` element will be focused during the first mount.
   */
  InputLabelProps: PropTypes.object,
  /**
   * @ignore
   */
  InputProps: PropTypes.object,
  /**
   * Override or extend the styles applied to the component.
   * See [CSS API](#css) below for more details.
   */
  SelectProps: PropTypes.object,
  /**
   * @ignore
   */
  autoComplete: PropTypes.string,
  /**
   * The color of the component. It supports those theme colors that make sense for this component.
   */
  autoFocus: PropTypes.bool,
  /**
   * The default value of the `input` element.
   */
  children: PropTypes.node,
  /**
   * If `true`, the `input` element will be disabled.
   */
  className: PropTypes.string,
  /**
   * If `true`, the label will be displayed in an error state.
   */
  classes: PropTypes.object,
  /**
   * Props applied to the [`FormHelperText`](/api/form-helper-text/) element.
   */
  color: PropTypes.oneOf(["primary", "secondary"]),
  /**
   * If `true`, the input will take up the full width of its container.
   */
  defaultValue: PropTypes.any,
  /**
   * The helper text content.
   */
  disabled: PropTypes.bool,
  /**
   * @ignore
   */
  error: PropTypes.bool,
  /**
   * The id of the `input` element.
   * Use this prop to make `label` and `helperText` accessible for screen readers.
   */
  fullWidth: PropTypes.bool,
  /**
   * Props applied to the [`InputLabel`](/api/input-label/) element.
   */
  helperText: PropTypes.node,
  /**
   * Props applied to the Input element.
   * It will be a [`FilledInput`](/api/filled-input/),
   * [`OutlinedInput`](/api/outlined-input/) or [`Input`](/api/input/)
   * component depending on the `variant` prop value.
   */
  hiddenLabel: PropTypes.bool,
  /**
   * [Attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Attributes) applied to the `input` element.
   */
  id: PropTypes.string,
  /**
   * Pass a ref to the `input` element.
   */
  inputProps: PropTypes.object,
  /**
   * The label content.
   */
  inputRef: refType,
  /**
   * If `dense` or `normal`, will adjust vertical spacing of this and contained components.
   */
  label: PropTypes.node,
  /**
   * If `true`, a textarea element will be rendered instead of an input.
   */
  margin: PropTypes.oneOf(["none", "dense", "normal"]),
  /**
   * Name attribute of the `input` element.
   */
  multiline: PropTypes.bool,
  /**
   * @ignore
   */
  name: PropTypes.string,
  /**
   * Callback fired when the value is changed.
   *
   * @param {object} event The event source of the callback.
   * You can pull out the new value by accessing `event.target.value` (string).
   */
  onBlur: PropTypes.func,
  /**
   * @ignore
   */
  onChange: PropTypes.func,
  /**
   * The short hint displayed in the input before the user enters a value.
   */
  onFocus: PropTypes.func,
  /**
   * If `true`, the label is displayed as required and the `input` element` will be required.
   */
  placeholder: PropTypes.string,
  /**
   * Number of rows to display when multiline option is set to true.
   */
  required: PropTypes.bool,
  /**
   * Maximum number of rows to display when multiline option is set to true.
   */
  rows: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /**
   * Render a [`Select`](/api/select/) element while passing the Input element to `Select` as `input` parameter.
   * If this option is set you must pass the options of the select as children.
   */
  rowsMax: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /**
   * Props applied to the [`Select`](/api/select/) element.
   */
  select: PropTypes.bool,
  /**
   * The size of the text field.
   */
  size: PropTypes.oneOf(["small", "medium"]),
  /**
   * Type of the `input` element. It should be [a valid HTML5 input type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Form_%3Cinput%3E_types).
   */
  type: PropTypes.string,
  /**
   * The value of the `input` element, required for a controlled component.
   */
  value: PropTypes.any,
  /**
   * The variant to use.
   */
  variant: PropTypes.oneOf(["standard", "outlined", "filled"])
};
/* eslint-enable react/boolean-prop-naming */

export default TextField;
