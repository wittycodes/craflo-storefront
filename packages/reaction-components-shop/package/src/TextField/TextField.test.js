import React from "react";
import { MenuItem } from "@material-ui/core";
import { render } from "../tests/index.js";
import TextField from "./TextField";

test("snapshot - singleline", () => {
  const { asFragment } = render(<TextField value="hello" />);
  expect(asFragment()).toMatchSnapshot();
});

test("snapshot - singleline - disabled", () => {
  const { asFragment } = render(<TextField value="hello" disabled />);
  expect(asFragment()).toMatchSnapshot();
});

test("snapshot - singleline - error", () => {
  const { asFragment } = render(<TextField value="hello" error helpText="Help text" />);
  expect(asFragment()).toMatchSnapshot();
});

test("snapshot - multiline", () => {
  const { asFragment } = render(<TextField value="hello" multiline />);
  expect(asFragment()).toMatchSnapshot();
});

test("snapshot - multiline - expanded", () => {
  const { asFragment } = render(<TextField value="hello" multiline rows={4} />);
  expect(asFragment()).toMatchSnapshot();
});

test("snapshot - multiline - disabled", () => {
  const { asFragment } = render(<TextField value="hello" multiline disabled />);
  expect(asFragment()).toMatchSnapshot();
});

test("snapshot - multiline - error state", () => {
  const { asFragment } = render(<TextField value="hello" multiline error helpText="Help text" />);
  expect(asFragment()).toMatchSnapshot();
});

test("snapshot - select", () => {
  const { asFragment } = render((
    <TextField
      value={1}
      select
    >
      <MenuItem value={1}>Option 1</MenuItem>
      <MenuItem value={2}>Option 2</MenuItem>
      <MenuItem value={3}>Option 3</MenuItem>
    </TextField>
  ));
  expect(asFragment()).toMatchSnapshot();
});

test("snapshot - select - disabled", () => {
  const { asFragment } = render((
    <TextField
      value={1}
      select
      disabled
    >
      <MenuItem value={1}>Option 1</MenuItem>
      <MenuItem value={2}>Option 2</MenuItem>
      <MenuItem value={3}>Option 3</MenuItem>
    </TextField>
  ));
  expect(asFragment()).toMatchSnapshot();
});

test("snapshot - select - error state", () => {
  const { asFragment } = render((
    <TextField
      value={1}
      select
      error
    >
      <MenuItem value={1}>Option 1</MenuItem>
      <MenuItem value={2}>Option 2</MenuItem>
      <MenuItem value={3}>Option 3</MenuItem>
    </TextField>
  ));
  expect(asFragment()).toMatchSnapshot();
});
