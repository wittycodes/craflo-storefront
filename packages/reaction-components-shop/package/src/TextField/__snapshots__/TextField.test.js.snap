// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot - multiline - disabled 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root makeStyles-root-291 Mui-disabled Mui-disabled makeStyles-disabled-294 MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline makeStyles-multiline-296"
    >
      <textarea
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input makeStyles-input-298 Mui-disabled Mui-disabled makeStyles-disabled-294 MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
        disabled=""
        rows="1"
        style="height: 0px; overflow: hidden;"
      >
        hello
      </textarea>
      <textarea
        aria-hidden="true"
        class="MuiInputBase-input MuiOutlinedInput-input makeStyles-input-298 Mui-disabled Mui-disabled makeStyles-disabled-294 MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
        readonly=""
        style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
        tabindex="-1"
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-345 MuiOutlinedInput-notchedOutline makeStyles-notchedOutline-297"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-346"
          style="width: 0.01px;"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`snapshot - multiline - error state 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    helptext="Help text"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root makeStyles-root-349 Mui-error Mui-error makeStyles-error-351 MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline makeStyles-multiline-354"
    >
      <textarea
        aria-invalid="true"
        class="MuiInputBase-input MuiOutlinedInput-input makeStyles-input-356 MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
        rows="1"
        style="height: 0px; overflow: hidden;"
      >
        hello
      </textarea>
      <textarea
        aria-hidden="true"
        class="MuiInputBase-input MuiOutlinedInput-input makeStyles-input-356 MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
        readonly=""
        style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
        tabindex="-1"
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-403 MuiOutlinedInput-notchedOutline makeStyles-notchedOutline-355"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-404"
          style="width: 0.01px;"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`snapshot - multiline - expanded 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root makeStyles-root-233 MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline makeStyles-multiline-238"
    >
      <textarea
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input makeStyles-input-240 MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
        rows="4"
      >
        hello
      </textarea>
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-287 MuiOutlinedInput-notchedOutline makeStyles-notchedOutline-239"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-288"
          style="width: 0.01px;"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`snapshot - multiline 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root makeStyles-root-175 MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline makeStyles-multiline-180"
    >
      <textarea
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input makeStyles-input-182 MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
        rows="1"
        style="height: 0px; overflow: hidden;"
      >
        hello
      </textarea>
      <textarea
        aria-hidden="true"
        class="MuiInputBase-input MuiOutlinedInput-input makeStyles-input-182 MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
        readonly=""
        style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
        tabindex="-1"
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-229 MuiOutlinedInput-notchedOutline makeStyles-notchedOutline-181"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-230"
          style="width: 0.01px;"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`snapshot - select - disabled 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root makeStyles-root-488 Mui-disabled Mui-disabled makeStyles-disabled-491 MuiInputBase-fullWidth MuiInputBase-formControl"
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby=" "
        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input makeStyles-input-495 Mui-disabled Mui-disabled makeStyles-disabled-491 Mui-disabled"
        role="button"
      >
        Option 1
      </div>
      <input
        type="hidden"
        value="1"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-565 MuiOutlinedInput-notchedOutline makeStyles-notchedOutline-494"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-566"
          style="width: 0.01px;"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`snapshot - select - error state 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root makeStyles-root-569 Mui-error Mui-error makeStyles-error-571 MuiInputBase-fullWidth MuiInputBase-formControl"
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby=" "
        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input makeStyles-input-576"
        role="button"
        tabindex="0"
      >
        Option 1
      </div>
      <input
        type="hidden"
        value="1"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-646 MuiOutlinedInput-notchedOutline makeStyles-notchedOutline-575"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-647"
          style="width: 0.01px;"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`snapshot - select 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root makeStyles-root-407 MuiInputBase-fullWidth MuiInputBase-formControl"
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby=" "
        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input makeStyles-input-414"
        role="button"
        tabindex="0"
      >
        Option 1
      </div>
      <input
        type="hidden"
        value="1"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-484 MuiOutlinedInput-notchedOutline makeStyles-notchedOutline-413"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-485"
          style="width: 0.01px;"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`snapshot - singleline - disabled 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root makeStyles-root-59 Mui-disabled Mui-disabled makeStyles-disabled-62 MuiInputBase-fullWidth MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input makeStyles-input-66 Mui-disabled Mui-disabled makeStyles-disabled-62"
        disabled=""
        type="text"
        value="hello"
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-113 MuiOutlinedInput-notchedOutline makeStyles-notchedOutline-65"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-114"
          style="width: 0.01px;"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`snapshot - singleline - error 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    helptext="Help text"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root makeStyles-root-117 Mui-error Mui-error makeStyles-error-119 MuiInputBase-fullWidth MuiInputBase-formControl"
    >
      <input
        aria-invalid="true"
        class="MuiInputBase-input MuiOutlinedInput-input makeStyles-input-124"
        type="text"
        value="hello"
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-171 MuiOutlinedInput-notchedOutline makeStyles-notchedOutline-123"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-172"
          style="width: 0.01px;"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`snapshot - singleline 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root makeStyles-root-1 MuiInputBase-fullWidth MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input makeStyles-input-8"
        type="text"
        value="hello"
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-55 MuiOutlinedInput-notchedOutline makeStyles-notchedOutline-7"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-56"
          style="width: 0.01px;"
        >
          <span>
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
